Hotfix/remove username from app user

oreze opened the pull request:
Separated the address entity to the user entity

oreze committed:
In the middle of refactoring addresses for user

oreze committed:
Prepare rest of the app for new way to add address

- fix tests
- refactor validators, dtos and other stuff

oreze committed:
Add new validation rules to ApplicationUserValidator

oreze committed:
Add new migrations for Identity service

oreze committed:
Fix migrations, add some stuff

oreze committed:
Add setting user address during adding new one

- add userId argument to IAddressRepository
- add SetUserAddress method in user aggregate root
- fix seeding database with addresses

RedRabel99 reviewed (changes requested):
Request on auth/register endpoint throws System.InvalidOperationException.
Should be fixed before merging.

KSi3m reviewed (changes requested):
There is an issue with address in registration. Not really sure where the problem lies

oreze committed:
Fix registering user with or without address

- add POSTGRES_DB parameter to docker-compose for local development
- add creating address without attaching it to the user
- fix seeding identity context
- create new migrations
- add try-catch clause in register for validation purposes

RedRabel99 reviewed (approved):
Registration works now correctly.
